tools/azure-pipelines-tasks-json-generate.mjs

```diff
@@ -0,0 +1,60 @@
+import os from 'os';
+import { promisify } from 'util';
+import fs from 'fs-extra';
+import g from 'glob';
+import shell from 'shelljs';
+import Git from 'simple-git';
+import path from 'upath';
+
+const glob = promisify(g);
+
+const localPath = path.join(os.tmpdir(), 'azure-pipelines-tasks');
+
+await (async () => {
+  await fs.ensureDir(localPath);
+  const git = Git(localPath);
+
+  if (await git.checkIsRepo()) {
+    await git.pull();
+  } else {
+    await git.clone(
+      'https://github.com/microsoft/azure-pipelines-tasks.git',
+      '.'
+    );
+  }
+
+  const files = (await glob(path.join(localPath, '**/task.json'))).map((file) =>
+    file.replace(`${localPath}/`, '')
+  );
+
+  /** @type {Record<string, Set<string>>} */
+  const tasks = {};
+
+  for (const file of files) {
+    const revs = (await git.raw(['rev-list', 'master', '--', file])).split(
```

Comment:
we should detect main branch if it's changed in future

Reply:
`git rev-list HEAD` gives equal results and sidesteps the issue. I assume `git clone` will pull the default branch.